src/fal/cli/dbt_model_generator.py

```diff
@@ -0,0 +1,132 @@
+import ast
+import astor
+import re
+from typing import List, Union
+from pathlib import Path
+from more_itertools import flatten
+
+from faldbt.project import FalDbt
+
+
+SQL_MODEL_TEMPLATE = """
+{{ config(materialized='ephemeral') }}
+/*
+FAL_GENERATED
+
+Script dependencies:
+
+__deps__
+
+*/
+
+SELECT * FROM {{ target.schema }}.{{ model.name }}
+"""
+
+
+def generate_python_dbt_models(fal_dbt: FalDbt):
+    python_paths = _find_python_files(fal_dbt.source_paths)
+
+    for py_path in python_paths:
+        sql_path = py_path.with_suffix(".sql")
+        _check_path_safe_to_write(sql_path, py_path)
+
+        with open(py_path) as file:
+            module = ast.parse(file.read(), str(py_path), "exec")
+
+        dbt_deps: str = _generate_dbt_dependencies(module)
+        sql_contents = SQL_MODEL_TEMPLATE.replace("__deps__", dbt_deps)
+
+        with open(sql_path, "w") as file:
+            file.write(sql_contents)
+
+
+def _check_path_safe_to_write(sql_path: Path, py_path: Path):
+    if sql_path.exists():
+        with open(sql_path, "r") as file:
+            contents = file.read()
+            if "FAL_GENERATED" not in contents:
+                raise RuntimeError(
+                    f"File '{sql_path}' not generated by fal would be overwritten by generated model of '{py_path}'. Please delete or move."
+                )
+
+
+def _find_python_files(source_dirs: Union[List[str], List[Path]]) -> List[Path]:
+    return list(
+        flatten(
+            map(
+                lambda d: (p for p in Path(d).rglob("*.py") if p.is_file()),
```

Comment:
Should define this lambda as a separate function? We will also need to process .ipynb files.

Reply:
Not sure if we will process ipynb as models or just as hooks.